### ISSUE TITLE
Write and publish a guide on how to deploy to a new DigitalOcean droplet

### ISSUE THREAD

**Issue by adrienjoly** [openwhyd/openwhyd#172]
Cf Kamal's email:

> create a guide on how to deploy their project on DigitalOcean or write a blog post on how they are using DigitalOcean to host their project.

> I saw that you use Docker, so it can be a short guide on spinning up a Droplet using the Docker one-click image and setting up the containers.

**Comment by uLan08** [comment]
Hi, I would like to work on this.

**Comment by adrienjoly** [comment]
Hi! Sure, thanks for proposing your contribution!

Do you know how you're going to proceed?

Don't hesitate if you have any question and/or would like some guidance while getting started on this.

**Comment by uLan08** [comment]
Yes I have an idea on how to proceed. I would most probably just try setting up on a droplet and base the documentation on how I did it. Also you said

> Make Openwhyd's repo deployable in one click (see https://www.digitalocean.com/products/one-click-apps/), e.g. by using our docker image (see our Dockerfile)

But the way I understood 

> I saw that you use Docker, so it can be a short guide on spinning up a Droplet using the Docker one-click image and setting up the containers.

Is to use the docker one click app and run the openwhyd images and not necessarily creating an openwhyd one click app.

**Comment by uLan08** [comment]
With my initial research. It seems that creating one click apps for Digital Ocean is not currently possible. Digital Ocean sets this up internally. 

https://www.digitalocean.com/community/questions/how-do-i-make-a-1-click-installed-application

This was posted way back in 2014 but I still don't see any documentation for this. 

I would just now then utilize the Docker one click app and the docker-compose file of the project.

**Comment by uLan08** [comment]
Woah, actually this "custom image" tab is the way to create a custom one click app. I totally overlooked this one. Looks like this feature was just released last September 25. 

Anyway, I took a look at it and it maybe too new for me at the moment. This would require creating a Linux image and not just a Docker image, I haven't done something like this before. I might just use the Docker one click app and deploy an openwhyd Docker image for now. We could start off here and later on we could get back at it and try creating an openwhyd Linux image, if you're okay with that. 

I currently have some free credits so you don't have to send me some.

**Comment by adrienjoly** [comment]
Great news:
- `openwhyd/openwhyd` is now available on https://hub.docker.com/r/openwhyd/openwhyd/tags/
- I followed [@uLan08's deploy guide](https://github.com/openwhyd/openwhyd/pull/176/files#diff-10e195dfdc7f12b07c3e1eb8b7bfeaaa), and successfully made Openwhyd work on a $5/month droplet! (1 vCPU + 1 GB of RAM. It would not scale but it does run!) 🎉

**Comment by adrienjoly** [comment]
FYI, Kamal from DigitalOcean sent me this feedback + explanations:

> The docs look great. Providing a docker compose file with instructions for the one-click Docker image is probably the easiest way to go about it.
>
> The recently released Custom Images feature that you've mentioned in the issue is for uploading pre-built Linux images to create Droplets off of. Right now it's not possible to share an image like that with other users but that's a feature we're exploring.

So I think that we're doing it right 💪

**Comment by uLan08** [comment]
Thanks as well @adrienjoly ! 

I would be interested in contributing again. I haven't looked at other issues yet but my main interests would probably be backend or ops related work. And as I have mentioned, I can also provide help with #165 if needed.

**Comment by adrienjoly** [comment]
Awesome!

I think that Brawrdon is on it, so I guess that you could offer your help. Or take over if he no longer wants to do it.

Otherwise, if you're into back-end, there's a project that may be interesting for you and that on which your contribution could have a huge impact on Openwhyd's future: find a way for external clients/front-ends to auth & connect to Openwhyd's API. (see issue #55)

I was think of adding a OAuth layer to Openwhyd's back-end, but we're open to other solutions, as long as we don't make too many changes to the current back-end codebase. Ideally, the proposed solution would either *wrap around* or *proxy* Openwhyd's existing API.

How does that sounds? I'd love to read your ideas and suggestions in issue #55, if you have any! ^^

**Comment by uLan08** [comment]
This looks interesting! I have read your initial discussions and for now I agree with using a wrapper to handle the OAuth process, but I'll have to dive in deeper and see. I'll let you know 
what I think then.
